### PR TITLE
DMEERX-567 DTR rule set for Plan of Care For Home Health Service

### PR DESCRIPTION
**Pull request by yuliumitre**
This PR creates the plan of care and certification form for Home Health Service

To test this PR, please use the PR from test-ehr repo, https://github.com/HL7-DaVinci/test-ehr/pull/30
CRD PR: https://github.com/HL7-DaVinci/CRD/pull/210
The patient is pat014 - Roosevelt Theodore

Note: My PR was originally created based on Howard's branch for Home Health Service CRD rule (but that PR hasn't been merged yet) and my branch sort diverged from there. So please ignore anything in Questionnaire-R4-HomeHealthServices.json and HomeHealthServicesPrepopulation-0.0.1.cql. 

There are several sections need clarification and they are not populated or partially done:
- Functional assessment
- Description of risk for emergency department visits and hospital readmission and all necessary
interventions to address risk
- Patient and caregiver education and training to facilitate timely discharge:
- Patient-specific interventions and education
- Discharge plan
- Preparer signature
- Revisions of the POC communicated to

### CODE REVIEW

File: Shared/R4/files/BasicClinicalInfoPrepopulation-0.1.0.cql [hunk 1]
```diff
@@ -29,3 +29,61 @@ define function CodesFromAllergies(AllergyList List<AllergyIntolerance>):
...
+  distinct(
+    flatten(
+      CondList C
+        let DiagnosesCodings:
+          (C.code.coding) CODING where CODING.system.value in {
+            'http://hl7.org/fhir/sid/icd-10',
+            'http://hl7.org/fhir/sid/icd-10-cm',
+            'http://snomed.info/sct'
+          }
+          return Tuple {
```
Comment (yunwwang) [inline Shared/R4/files/BasicClinicalInfoPrepopulation-0.1.0.cql]: using FHIRHelpers.ToCode

File: Shared/R4/files/BasicClinicalInfoPrepopulation-0.1.0.cql [hunk 2]
```diff
@@ -29,3 +29,61 @@ define function CodesFromAllergies(AllergyList List<AllergyIntolerance>):
...
+    )
+  )  
+define "AllProcedures": ProcedureCoding([Procedure])
+define function ProcedureCoding(ProcedureList List<FHIR.Procedure>):
+  distinct(
+    flatten(
+      ProcedureList P
+        let ProcedureCodings:
+          (P.code.coding) CODING 
+          return Tuple {
```
Comment (yunwwang) [inline Shared/R4/files/BasicClinicalInfoPrepopulation-0.1.0.cql]: Use FHIRHelpers.ToCode

File: Shared/R4/files/BasicClinicalInfoPrepopulation-0.1.0.cql [hunk 3]
```diff
@@ -29,3 +29,61 @@ define function CodesFromAllergies(AllergyList List<AllergyIntolerance>):
...
+  define "True": true
+
+  define "MedicationList": GetMedicationsList([MedicationStatement])
+  define function GetMedicationsList(MedStatementList List<MedicationStatement>):
+    distinct(
+      flatten(
+        MedStatementList MedStatement
+          let MedicationList:
+            (MedStatement.medication.coding) CODING
+            return Tuple {
```
Comment (yunwwang) [inline Shared/R4/files/BasicClinicalInfoPrepopulation-0.1.0.cql]: Use FHIRHelpers.ToCode

File: HomeHealthServices/R4/files/HomeHealthServicesPlanOfCarePrepopulation-0.0.1.cql [hunk 4]
```diff
@@ -0,0 +1,125 @@
+library HomeHealthServicesPlanOfCarePrepopulation version '0.0.1'
+using FHIR version '4.0.0'
+include FHIRHelpers version '4.0.0' called FHIRHelpers
+
+codesystem "ICD-10-CM": 'http://hl7.org/fhir/sid/icd-10-cm'
```
Comment (yunwwang) [inline HomeHealthServices/R4/files/HomeHealthServicesPlanOfCarePrepopulation-0.0.1.cql]: Are these Code Systems used in this CQL?

File: HomeHealthServices/R4/files/HomeHealthServicesPlanOfCarePrepopulation-0.0.1.cql [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+define "RelavantDiagnosisCoding": ConvertConditionDetails(RelavantDiagnosis)
+
+define function ConvertConditionDetails(CondList List<Condition>):
+  distinct(
+    flatten(
+      CondList C
+        let ConditionCodings:
+          (C.code.coding) CODING
+          return Tuple {
```
Comment (yunwwang) [inline HomeHealthServices/R4/files/HomeHealthServicesPlanOfCarePrepopulation-0.0.1.cql]: Use FHIRHelpers.ToCode

File: HomeHealthServices/R4/files/HomeHealthServicesPrepopulation-0.0.1.cql [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+codesystem "LOINC": 'http://loinc.org'
+codesystem "SNOMED-CT": 'http://snomed.info/sct'
+
+//define Today: Today()
+
+parameter service_request ServiceRequest
+
+context Patient
+
+//define "ServiceRequestHcpcsCoding": singleton from (
```
Comment (yunwwang) [inline HomeHealthServices/R4/files/HomeHealthServicesPrepopulation-0.0.1.cql]: Remove not used comments

File: HomeHealthServices/R4/resources/Questionnaire-R4-HomeHealthServices.json [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+            "code": "other",
+            "display": "Other"
+          }
+        ]
+      }
+    }
+  ],
+  "item": [
+    {
+      "linkId": "1",
```
Comment (yunwwang) [inline HomeHealthServices/R4/resources/Questionnaire-R4-HomeHealthServices.json:58]: Use PatientInfo questionnaire

File: HomeHealthServices/R4/resources/Questionnaire-R4-HomeHealthServices.json [hunk 8]
```diff
@@ -0,0 +1,238 @@
...
+                "language": "text/cql",
+                "expression": "\"BasicPatientInfoPrepopulation\".MedicareId"
+              }
+            }
+          ]
+        }
+      ]
+    },
+    {
+      "linkId": "2",
```
Comment (yunwwang) [inline HomeHealthServices/R4/resources/Questionnaire-R4-HomeHealthServices.json:156]: Use PractitionerInfo questionnaire

File: HomeHealthServices/R4/resources/Questionnaire-R4-HomeHealthServicesPlanOfCare.json [hunk 9]
```diff
@@ -0,0 +1,618 @@
...
+  "resourceType": "Questionnaire",
+  "id": "HomeHealthServicesPlanOfCare",
+  "name": "HomeHealthServicesPlanOfCare",
+  "meta": {
+    "profile": [
+      "http://hl7.org/fhir/StructureDefinition/cqf-questionnaire",
+      "http://hl7.org/fhir/us/davinci-dtr/StructureDefinition/dtr-questionnaire-r4"
+    ]
+  },
+  "extension": [
```
Comment (yunwwang) [inline HomeHealthServices/R4/resources/Questionnaire-R4-HomeHealthServicesPlanOfCare.json:11]: Remove not used cqf-libraries

File: HomeHealthServices/R4/resources/Library-R4-HomeHealthServicesPlanOfCare-prepopulation.json [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+      }
+    }
+  ],
+  "dataRequirement": [
+    {
+      "type": "Procedure",
+      "codeFilter": [
+          {
+              "path": "code",
+              "valueSet": "http://cts.nlm.nih.gov/fhir/ValueSet/2.16.840.1.113762.1.4.1219.90"
```
Comment (yunwwang) [inline HomeHealthServices/R4/resources/Library-R4-HomeHealthServicesPlanOfCare-prepopulation.json]: Is this ValueSet used in CQL?

File: HomeHealthServices/R4/resources/Questionnaire-R4-HomeHealthServices.json [hunk 11]
```diff
@@ -0,0 +1,238 @@
...
+          },
+          {
+            "code": "other",
+            "display": "Other"
+          }
+        ]
+      }
+    }
+  ],
+  "item": [
```
Comment (yunwwang) [inline HomeHealthServices/R4/resources/Questionnaire-R4-HomeHealthServices.json:56]: Does this Questionnaire have only Patient and Practitioner section?

File: HomeHealthServices/R4/resources/Questionnaire-R4-HomeHealthServices.json (hunk 7 above)
Comment (yuliumitre) [reply HomeHealthServices/R4/resources/Questionnaire-R4-HomeHealthServices.json:58]: @yunwwang This questionnaire is WIP and Howard is still working on it. My PR was originally created based on his branch and sort diverged from there. So please ignore anything in Questionnaire-R4-HomeHealthServices.json and HomeHealthServicesPrepopulation-0.0.1.cql. I should have indicated in the PR description. Sorry about that.

File: HomeHealthServices/R4/resources/Questionnaire-R4-HomeHealthServices.json (hunk 8 above)
Comment (yuliumitre) [reply HomeHealthServices/R4/resources/Questionnaire-R4-HomeHealthServices.json:156]: Same as above.

File: HomeHealthServices/R4/files/HomeHealthServicesPlanOfCarePrepopulation-0.0.1.cql (hunk 4 above)
Comment (yuliumitre) [reply HomeHealthServices/R4/files/HomeHealthServicesPlanOfCarePrepopulation-0.0.1.cql]: Good catch. Removed.

File: HomeHealthServices/R4/resources/Questionnaire-R4-HomeHealthServices.json (hunk 11 above)
Comment (yuliumitre) [reply HomeHealthServices/R4/resources/Questionnaire-R4-HomeHealthServices.json:56]: Howard is still working on this.

**Review by yunwwang — APPROVED** [review]

**Pull request merged**